anchor/network/src/keypair_utils.rs

```diff
@@ -0,0 +1,51 @@
+use libp2p::identity::{secp256k1, Keypair};
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::PathBuf;
+use tracing::{debug, warn};
+
+pub const NETWORK_KEY_FILENAME: &str = "key";
+
+/// Loads a private key from disk. If this fails, a new key is
+/// generated and is then saved to disk.
+///
+/// Currently only secp256k1 keys are allowed, as these are the only keys supported by discv5.
+pub fn load_private_key(network_dir: &PathBuf) -> Keypair {
+    // check for key from disk
+    let network_key_f = network_dir.join(NETWORK_KEY_FILENAME);
+    if let Ok(mut network_key_file) = File::open(network_key_f.clone()) {
+        let mut key_bytes: Vec<u8> = Vec::with_capacity(36);
+        match network_key_file.read_to_end(&mut key_bytes) {
+            Err(_) => debug!("Could not read network key file"),
```

Comment:
Good question - the current behaviour is, if the file does not exist, is invalid or cannot be read, then we generate a new one. 

This is the same behaviour from Lighthouse - tbh I didn't really think about changing it when I copied this from LH. This node private key is mostly transparent to the user and used for node id, peer id and  noise protocol encryption. As it's transparent to the user, I think it's better UX to treat it as invalid if the file cannot be read, and regenerate a new one, instead of crashing and asking the user to fix it manually or delete the existing corrupted file.